test/acceptance/block_sync_internal_test.go

```diff
@@ -0,0 +1,134 @@
+
+import (
+	"context"
+	"github.com/orbs-network/orbs-network-go/crypto/digest"
+	"github.com/orbs-network/orbs-network-go/test/builders"
+	"github.com/orbs-network/orbs-network-go/test/harness"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/client"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestInternalBlockSync_TransactionPool(t *testing.T) {
+
+	blockCount := primitives.BlockHeight(10)
+	txBuilders := make([]*builders.TransactionBuilder, blockCount)
+	for i := 0; i < int(blockCount); i++ {
+		txBuilders[i] = builders.TransferTransaction().WithAmountAndTargetAddress(uint64(i+1)*10, builders.AddressForEd25519SignerForTests(6))
+	}
+
+	harness.Network(t).
+		AllowingErrors(
+			"leader failed to save block to storage",              // (block already in storage, skipping) TODO investigate and explain, or fix and remove expected error
+			"internal-node sync to consensus algo failed",         //TODO Remove this once internal node sync is implemented
+			"all consensus 0 algos refused to validate the block", //TODO investigate and explain, or fix and remove expected error
+			"all consensus 1 algos refused to validate the block", //TODO investigate and explain, or fix and remove expected error
+		).
+		WithSetup(func(ctx context.Context, network harness.TestNetworkDriver) {
+			for i := primitives.BlockHeight(1); i <= blockCount; i++ {
+				blockPair := builders.BenchmarkConsensusBlockPair().
+					WithTransaction(txBuilders[i-1].Build()).
+					WithReceiptsForTransactions().
+					WithHeight(i).
+					Build()
+				network.BlockPersistence(0).WriteNextBlock(blockPair)
+			}
+		}).Start(func(ctx context.Context, network harness.TestNetworkDriver) {
+
+		lastTx := txBuilders[len(txBuilders)-1].Build().Transaction()
+		waitForTransactionStatusCommitted(network, ctx, lastTx, 0)
+		waitForTransactionStatusCommitted(network, ctx, lastTx, 1)
+
+		// Resend an already committed transaction to Leader
+		leaderTxResponse, ok := <-network.SendTransaction(ctx, txBuilders[0].Builder(), 0)
+		require.True(t, ok)
+		require.Equal(t, protocol.TRANSACTION_STATUS_DUPLICATE_TRANSACTION_ALREADY_COMMITTED, leaderTxResponse.TransactionStatus())
+
+		// Resend an already committed transaction to Non-Leader
+		nonLeaderTxResponse, ok := <-network.SendTransaction(ctx, txBuilders[0].Builder(), 1)
+		require.True(t, ok)
+		require.Equal(t, protocol.TRANSACTION_STATUS_DUPLICATE_TRANSACTION_ALREADY_COMMITTED, nonLeaderTxResponse.TransactionStatus())
+	})
+}
+
+func waitForTransactionStatusCommitted(network harness.TestNetworkDriver, ctx context.Context, lastTx *protocol.Transaction, nodeIndex int) {
+	var txStatusOut *services.GetTransactionStatusOutput
+	for txStatusOut == nil || txStatusOut.ClientResponse.TransactionStatus() != protocol.TRANSACTION_STATUS_COMMITTED {
```

Comment:
isn't this going to overwork the cpu?

Reply:
I can either use test.Eventually or add a  sleep(1*Millisecond) to the loop. Do you have a better idea how to wait for a transaction to reach TxPool?

Reply:
Used test.Eventually